html/includes/graphs/multiport/auth.inc.php

```diff
@@ -1,10 +1,8 @@
 <?php
 
-$auth = true;
-
 foreach (explode(',', $vars['id']) as $ifid) {
-    if (!$auth && !port_permitted($ifid)) {
-        $auth = false;
+    if ($auth || port_permitted($ifid)) {
+        $auth = true;
```

Comment:
These both seem wrong.  Only should be allowed if $auth or all ports are permitted.
```
if (!$auth) {
    foreach ($ids as $ifid) {
        if ($ifid) {
            $auth = true;
        } else {
            $auth = false;
            break;
        }
    }
}
```